classes/article/ArticleGalleyDAO.inc.php

```diff
@@ -451,6 +451,71 @@ function deleteAllPubIds($journalId, $pubIdType) {
 		}
 		$this->flushCache();
 	}
+
+	/**
+	 * Get all published article galleys with a pubId assigned and matching the specified settings.
+	 * @param $pubIdType string
+	 * @param $contextId integer optional
+	 * @param $title string optional
+	 * @param $author string optional
+	 * @param $issueId integer optional
+	 * @param $pubIdSettingName string optional
+	 * (e.g. medra::status or medra::registeredDoi)
+	 * @param $pubIdSettingValue string optional
+	 * @param $rangeInfo DBResultRange optional
+	 * @return DAOResultFactory
+	 */
+	function getByPubIdType($pubIdType, $contextId = null, $title = null, $author = null, $issueId = null, $pubIdSettingName = null, $pubIdSettingValue = null, $rangeInfo = null) {
+		$params = array();
+		if ($pubIdSettingName) {
+			$params[] = $pubIdSettingName;
+		}
+		$params[] = 'pub-id::'.$pubIdType;
+		if ($contextId) {
+			$params[] = (int) $contextId;
+		}
+		if ($title) {
+			$params[] = 'title';
+			$params[] = '%' . $title . '%';
+		}
+		if ($author) array_push($params, $authorQuery = '%' . $author . '%', $authorQuery, $authorQuery);
+		if ($issueId) {
+			$params[] = (int) $issueId;
+		}
+		import('classes.plugins.DOIPubIdExportPlugin');
+		if ($pubIdSettingName && $pubIdSettingValue && $pubIdSettingValue != DOI_EXPORT_STATUS_NOT_DEPOSITED) {
+			$params[] = $pubIdSettingValue;
+		}
+
+		$result = $this->retrieveRange(
+				'SELECT	g.*, sf.*
+			FROM	submission_galleys g
+				' . ($contextId?' JOIN submissions s ON (s.submission_id = g.submission_id)':'') . '
+				LEFT JOIN published_submissions ps ON (ps.submission_id = g.submission_id)
+				LEFT JOIN issues i ON (ps.issue_id = i.issue_id AND i.date_published IS NOT NULL)
+				LEFT JOIN submission_files sf ON (g.file_id = sf.file_id)
+				LEFT JOIN submission_files nsf ON (nsf.file_id = g.file_id AND nsf.revision > sf.revision AND nsf.file_id IS NULL )
+				LEFT JOIN submission_galley_settings gs ON (g.galley_id = gs.galley_id)
+				'. ($title?' LEFT JOIN submission_settings sst ON (s.submission_id = sst.submission_id)':'')
```

Comment:
You are right for the current locale -- I think it is ok to restrict it to the current locale, right?
I believe those sql locale checks for articles are wrong in general -- they look if there is current and primary locale, but the primary locale has no sense for an article, it doesn't necessarily have to be in the primary locale -- it has it's own submission locale.